### PR TITLE
Pariser parr

### PR DESCRIPTION
**Pull request by giovanni-br**
Fixing CI errors.

### CODE REVIEW

File: moha/rauk/PariserParr.py [hunk 1]
```diff
@@ -274,23 +274,43 @@ def compute_overlap(
...
-    return gamma
+    num_sites = len(u_onsite)
+    gamma_matrix = np.zeros((num_sites, num_sites))
+
+    for tpl in connectivity:
+        atom1, atom2 = tpl[0], tpl[1]
+        atom1_name, site1 = get_atom_type(atom1)
+        atom2_name, site2 = get_atom_type(atom2)
+
+        if site1 < num_sites and site2 < num_sites:
```
Comment (RichRick1) [inline moha/rauk/PariserParr.py:303]: I'm not sure if we need this check. Can't we have value of the `site` to be equal to number of sites? Let's say site 6 for the total of 6 sites?

**Comment by RichRick1** [comment]
The gh actions are failing because there is no file `affinity.json`. Changing the filename from `affinity.Json` to `affinity.json` should help

**Comment by PaulWAyers** [comment]
Can you specify where the data came from? Some of the values are not quite what I would have expected, but perhaps due to the source. Negative electron affinities are OK here, probably, I guess....it's a matter of taste as to what to do with negative electron affinities.

**Comment by giovanni-br** [comment]
> Can you specify where the data came from? Some of the values are not quite what I would have expected, but perhaps due to the source. Negative electron affinities are OK here, probably, I guess....it's a matter of taste as to what to do with negative electron affinities.

I took this data from https://en.wikipedia.org/wiki/Electron_affinity_(data_page). I am not sure about the truthfulness of the data...

**Comment by PaulWAyers** [comment]
That's fine. With that documented (via pull request and eventual package documentation) we are in good shape. We can make an executive decision on negative electron affinities later. Realistically, they mostly occur for elements that are not very interesting for PPP theory.

**Comment by giovanni-br** [comment]
> That's fine. With that documented (via pull request and eventual package documentation) we are in good shape. We can make an executive decision on negative electron affinities later. Realistically, they mostly occur for elements that are not very interesting for PPP theory.

Nice! I've already draft some documentation that explain this: https://github.com/theochem/ModelHamiltonian/blob/PariserParr/examples/rauk.ipynb

**Comment by PaulWAyers** [comment]
Not really. This is good enough for this purpose. We could use `AtomDB` but I think it's not worth the effort for this. Users can always propose their own constants if they want to. We should just be clear where we got the data from and on what day we downloaded it.

File: moha/hamiltonians.py [hunk 2]
```diff
@@ -703,3 +707,170 @@ def __init__(self,
...
+                 affinity_dct=None,
+                 Rxy_list=None,
+                 J_eq=None,
+                 J_ax=None):
+        r"""
+        Initialize t-J-U-V Hamiltonian.
+
+        Parameters
+        ----------
+        connectivity_ppp: Union[list, np.ndarray]
```
Comment (RichRick1) [inline moha/hamiltonians.py:735]: we need to make sure that docstring consistent with the parameters. For example, there is no parameter connectivity_ppp in the input parameters.

File: moha/hamiltonians.py [hunk 3]
```diff
@@ -703,3 +707,170 @@ def __init__(self,
...
+            List of tuples specifying sites and bonds
+            np.ndarray of shape (n_sites, n_sites)
+            between sites for the PPP model.
+        connectivity_heisenberg: np.ndarray
```
Comment (RichRick1) [inline moha/hamiltonians.py:739]: We need to make the docstring consistent with the the input parameters

File: moha/hamiltonians.py [hunk 4]
```diff
@@ -703,3 +707,170 @@ def __init__(self,
...
+            Dense or sparse matrix; default False
+        sym: int
+            Symmetry -- [2, 4, 8] default is 1
+
+        Returns
+        -------
+        scipy.sparse.csr_matrix or np.ndarray
+        """
+        two_body_ppp = self.ocupation_part.generate_two_body_integral(
+            basis, dense, sym)
```
Comment (RichRick1) [inline moha/hamiltonians.py:872]: It looks like order of parameters for generating two body hamiltonian for PPP and Heisenberg hamiltonians are inconsistent. We need to change order of Heisenberg model parameters and then change it in this line

File: moha/test/test_tjuv.py [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+                            sym=sym,
+                            J_eq=J_eq,
+                            J_ax=J_ax)
+
+    # Generate the zero body integral
+    tjuv_zero = tjuv_hamiltonian.generate_zero_body_integral()
+
+    heisenberg = HamHeisenberg(J_eq=J_eq, J_ax=J_ax, mu=0, connectivity=connectivity)
+    heisenberg_zero = heisenberg.generate_zero_body_integral()
+    hpp = HamPPP(connectivity=connectivity, alpha=alpha, beta=beta, gamma=np.zeros((2, 2)), charges=None, sym=None, u_onsite=[1, 1])
```
Comment (RichRick1) [inline moha/test/test_tjuv.py]: This test feels a bit fishy to me. The connectivity implies 6 site model. However, we provide `gamma` for 2 site system, and the same happening with `u_onsite`. I would expect this test to break. We need to investigate what's going on here

File: moha/test/test_tjuv.py [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+                               J_eq=J_eq,
+                               J_ax=J_ax,)
+
+    # Generate the one-body integral
+    tjuv_one_body = tjuv_hamiltonian.generate_one_body_integral(basis='spatial basis', dense=True)
+
+    heisenberg = HamHeisenberg(J_eq=J_eq, J_ax=J_ax, mu=0, connectivity=connectivity)
+    heisenberg_one_body = heisenberg.generate_one_body_integral(basis='spatial basis', dense=True)
+
+    hpp = HamPPP(connectivity=connectivity, alpha=alpha, beta=beta, gamma=np.zeros((2, 2)), charges=None, sym=None, u_onsite=u_onsite)
```
Comment (RichRick1) [inline moha/test/test_tjuv.py]: Another test, in which gamma is inconsistent with number of sites.

**Review by RichRick1 — CHANGES_REQUESTED** [review]
Great job! I especially like the last test when we test the energy spectrum of hamiltonian. However, there are some comments regarding the consistency of docstrings and parameter order. Also, some tests that I wouldn't expect to work are working. I left more detailed explanations in the in-line comments

File: moha/test/test_tjuv.py (hunk 5 above)
Comment (giovanni-br) [reply moha/test/test_tjuv.py]: I think this breaks! I have committed the changes.

File: moha/test/test_tjuv.py (hunk 6 above)
Comment (giovanni-br) [reply moha/test/test_tjuv.py]: Maybe it's because we are not checking if the size of gamma is correct in the two body term functions

**Review by RichRick1 — APPROVED** [review]
Looks great to me! Thanks for fixing the issue with PPP dict

**Pull request merged**